### ISSUE TITLE
Review Details: unknown name reported for pre-order printer

### ISSUE THREAD

**Issue by mindgraffiti** [woocommerce/woocommerce-ios#1644]
**Describe the bug**
Push notifications for a Review on a pre-ordered product show "Unknown" instead of the pre-ordered product name.

**To Reproduce**
1. On your test site, install the [Pre-Orders Plugin](https://woocommerce.com/products/woocommerce-pre-orders/). This is a plugin created by WC, so it should be supported in the app.
2. Create an order and check out with the pre-ordered product as the only item in your cart.
3. Wait for the push notification on the device. Tap it and wait to be navigated to the Reviews page.
4. Tap the review and note the "Unknown" title

**Screenshots**
<img src="https://user-images.githubusercontent.com/1062444/71104672-cc482780-2181-11ea-962f-1c761a15d3ad.PNG" width="350" />

**Expected behavior**
(Research needed). Is it possible to display the pre-ordered product's title?

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: any
- iOS version: any 
- WooCommerce iOS version: 3.3

**WordPress Environment**
<details>
```
Copy and paste the system status report from **WooCommerce > System Status** in WordPress admin.
```
</details>

**Comment by shiki** [comment]
Related to #1907

**Comment by astralbodies** [comment]
Resolved in #6367
